cmd/lncli/cmd_build_route.go

```diff
@@ -0,0 +1,94 @@
+// +build routerrpc
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/lightningnetwork/lnd"
+	"github.com/lightningnetwork/lnd/lnrpc/routerrpc"
+	"github.com/lightningnetwork/lnd/routing/route"
+	"github.com/urfave/cli"
+)
+
+var buildRouteCommand = cli.Command{
+	Name:     "buildroute",
+	Category: "Payments",
+	Usage:    "Build a route from a list hop pubkeys.",
+	Action:   actionDecorator(buildRoute),
+	Flags: []cli.Flag{
+		cli.Int64Flag{
+			Name: "amt",
+			Usage: "the amount to send expressed in satoshis. If" +
+				"not set, the minimum routable amount is used",
+		},
+		cli.Int64Flag{
+			Name: "final_cltv_delta",
+			Usage: "number of blocks the last hop has to reveal " +
+				"the preimage",
+			Value: lnd.DefaultBitcoinTimeLockDelta,
+		},
+		cli.StringFlag{
+			Name:  "hops",
+			Usage: "comma separated hex pubkeys",
+		},
+		cli.Uint64Flag{
+			Name: "outgoing_chan_id",
+			Usage: "short channel id of the outgoing channel to " +
+				"use for the first hop of the payment",
+			Value: 0,
+		},
+	},
+}
+
+func buildRoute(ctx *cli.Context) error {
+	conn := getClientConn(ctx, false)
+	defer conn.Close()
+
+	client := routerrpc.NewRouterClient(conn)
+
+	if !ctx.IsSet("hops") {
+		return errors.New("hops required")
+	}
+
+	// Build list of hop addresses for the rpc.
+	hops := strings.Split(ctx.String("hops"), ",")
+	rpcHops := make([][]byte, 0, len(hops))
+	for _, k := range hops {
+		pubkey, err := route.NewVertexFromStr(k)
+		if err != nil {
+			return fmt.Errorf("error parsing %v: %v", k, err)
+		}
+		rpcHops = append(rpcHops, pubkey[:])
+	}
+
+	var amt int64
+	amtSet := ctx.IsSet("amt")
+	if amtSet {
+		amt = ctx.Int64("amt") * 1000
```

Comment:
to avoid the confusion here, could you name one var `amt_sat` and the other `amt_msat`? 😅

Reply:
one of the vars is `amtSet`, not `amtSat` :smile: Renamed the other one to `amtMsat`. I didn't change the command line parameter, because our standard in most cases is implictly sats for `lncli`.